File: Fibonacci Number/fibonacci_number.py
# F(0) = 0, F(1) = 1

def fib(n):
    if n <= 0:
        return []
    elif n == 1:
        return [0]
    elif n == 2:
        return [0, 1]
    else:
        seq = fib(n - 1)
        seq.append(seq[-1] + seq[-2])
        return seq

n = int(input())
print(fib(n))

File: Fibonacci Number/fibonacci_number.cpp
class Solution {
public:
    int fib(int n) {
        if (n <= 1) return n;      // 基本情况：F(0)=0, F(1)=1
        return fib(n - 1) + fib(n - 2);  // 递归公式
    }
};
